crates/matrix-sdk-ui/src/app/mod.rs

```diff
@@ -74,22 +75,35 @@ impl App {
     /// Observe the current state of the application.
     ///
     /// See also [`AppState`].
-    pub fn observe_state(&self) -> impl Stream<Item = AppState> {
+    pub fn observe_state(&self) -> Subscriber<AppState> {
```

Comment:
Not exactly related to this PR, but I don't think we use the `observe_` prefix elsewhere, might make sense to remove?

Reply:
that's right; I've removed it.